tensorboardX/writer.py

```diff
@@ -768,24 +778,34 @@ def add_embedding(self, mat, metadata=None, label_img=None, global_step=None, ta
         # Maybe we should encode the tag so slashes don't trip us up?
         # I don't think this will mess us up, but better safe than sorry.
         subdir = "%s/%s" % (str(global_step).zfill(5), self._encode(tag))
-        save_path = os.path.join(self._get_file_writer().get_logdir(), subdir)
+        log_dir = self._get_file_writer().get_logdir()
+        save_path = os.path.join(log_dir, subdir)
+        s3_bucket = None
+        if save_path.startswith('s3://'):
+            s3_bucket = save_path.replace('s3://', '').split('/')[0]
+            save_path = save_path.replace('s3://', '/tmp/')
+            log_dir = log_dir.replace('s3://', '/tmp/')
         try:
             os.makedirs(save_path)
         except OSError:
             print(
                 'warning: Embedding dir exists, did you set global_step for add_embedding()?')
+        to_upload = []
         if metadata is not None:
             assert mat.shape[0] == len(
                 metadata), '#labels should equal with #data points'
-            make_tsv(metadata, save_path, metadata_header=metadata_header)
+            named_path = make_tsv(metadata, save_path, metadata_header=metadata_header)
+            upload_to_s3(s3_bucket, named_path)
         if label_img is not None:
             assert mat.shape[0] == label_img.shape[0], '#images should equal with #data points'
-            make_sprite(label_img, save_path)
+            named_path = make_sprite(label_img, save_path)
+            upload_to_s3(s3_bucket, named_path)
         assert mat.ndim == 2, 'mat should be 2D, where mat.size(0) is the number of data points'
-        make_mat(mat, save_path)
-        # new funcion to append to the config file a new embedding
-        append_pbtxt(metadata, label_img,
-                     self._get_file_writer().get_logdir(), subdir, global_step, tag)
+        named_path = make_mat(mat, save_path)
+        upload_to_s3(s3_bucket, named_path)
+        # new function to append to the config file a new embedding
+        named_path = append_pbtxt(metadata, label_img, log_dir, subdir, global_step, tag)
+        upload_to_s3(s3_bucket, named_path)
```

Comment:
Can we simply copy all files in `save_path` to there corresponding s3 bucket here? In this way, there is no need to touch embedding.py.